File: Dart/test/canAliceWin_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:leetcode/canAliceWin.dart';

void main() {
  test("test example", () {
    var input0 = [
      [1, 2, 3, 4, 10],
      [1, 2, 3, 4, 5, 14],
      [5, 5, 5, 25]
    ];
    var expected = [false, true, true];

    final sln = Solution();
    for (var i = 0; i < input0.length; i++) {
      var output = sln.canAliceWin(input0[i]);
      expect(output, expected[i]);
    }
  });
}

File: Typescript/test/canAliceWin.test.ts
import { expect, test } from '@jest/globals';
import * as solution from '../src/canAliceWin';

test('Example 1', () => {
	let input = [1, 2, 3, 4, 10];
	let expected = false;
	expect(solution.canAliceWin(input)).toEqual(expected);
});

test('Example 2', () => {
	let input = [1, 2, 3, 4, 5, 14];
	let expected = true;
	expect(solution.canAliceWin(input)).toEqual(expected);
});

test('Example 3', () => {
	let input = [5, 5, 5, 25];
	let expected = true;
	expect(solution.canAliceWin(input)).toEqual(expected);
});
